crittercism/models.py

```diff
@@ -82,16 +90,47 @@ def date_occurrences(self, date):
         return self.date_map.get(self.normalize_datetime_to_midnight(date), 0)
 
     def latest_occurrences_by_version(self):
-        todays_date = self._todays_date.strftime('%Y-%m-%d')
+        todays_date = self._todays_date.strftime('%Y%m%d')
         return {
             version: num_list[0] if date == todays_date else 0
             for version, [date, num_list]
-            in self._data[u'dailyOccurrencesByVersion'].items()
+            in self._data['daily_occurrences_by_version'].items()
         }
 
     def build_date_map(self):
-        incomplete_data_latest_date, daily_data = self._data[u'dailyOccurrencesByVersion'][u'total']
-        latest_complete_datetime = datetime.strptime(incomplete_data_latest_date, '%Y-%m-%d')
+        """
+        For crashes, this uses the array of arrays in 'daily_occourrences'
+        For exceptions, adds the 'dailyOccurrencesByVersion' for each version
+        to get the total daily occurrences.
+
+        :return: The datetime of the last day of complete data,
+        and a dictionary of datetimes with the total number of occurrences
+        of this crash on that day.
+        """
+        if self.is_crash:
+            incomplete_data_latest_date, daily_data = self._data['daily_occurrences']
+            latest_complete_datetime = datetime.strptime(
+                incomplete_data_latest_date, '%Y%m%d')
+        else:
+            daily_occurrences_by_version = self._data.get(u'dailyOccurrencesByVersion')
+            if daily_occurrences_by_version:
+                incomplete_data_latest_date, daily_data = daily_occurrences_by_version[u'total']
+                latest_complete_datetime = datetime.strptime(incomplete_data_latest_date, '%Y-%m-%d')
+                total = defaultdict(int)
+                for version, errors in daily_occurrences_by_version.items():
+                    start_date = datetime.strptime(errors[0], '%Y-%m-%d')
+                    daily_error_counts = errors[1]
+                    for index, error_count in enumerate(reversed(daily_error_counts)):
+                        date = (start_date - timedelta(days=index))
+                        total[date] += error_count
+
+                dates = sorted(total.keys())
+                total_daily_occurrences = (dates[-1].strftime('%Y-%m-%d'), [])
+
+                for date in dates:
+                    total_daily_occurrences[1].append(total[date])
+            else:
+                return None
```

Comment:
Yes, it should return a datetime object and a dictionary of datetimes and error totals.